Sync v2-10-stable with v2-10-test to release 2.10.1rc1

utkarsharma2 opened the pull request:
Time for 2.10.1rc1!

bbovenzi committed:
Fix try selector refresh (#41483) (#41503)

potiuk committed:
Remove debian bullseye support (#41568) (#41569)

(cherry picked from commit 5c323a9c562c6e886dbba460fc733a6f8590bf8b)

bbovenzi committed:
Field Deletion Warning when editing Connections (#41144) (#41504)

* able to change the 'Changed Row' display message after edit

* added message in connection form to warn of empty fields

* attempt to warn the specific fields cannot be empty

* revert change because need to check fields before save is clicked

* issues warning for specific fields that can't be deleted after save

* removed the individual warnings

* changed status to concise string

* added more concise suggestion



---------

Co-authored-by: Lucy Hu <90779522+lh5844@users.noreply.github.com>

uranusjr committed:
[Backport] Deprecate implicit default DAG schedule (#41469)

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

jscheffl committed:
Fix UI rendering when XCom is INT, FLOAT, BOOL or NULL (#41516) (#41605)

uranusjr committed:
Fix InletEventsAccessors type stub (#41572) (#41607)

Co-authored-by: phi-friday <phi.friday@gmail.com>

mobuchowski committed:
Set better logging level for path wrapper (#41615) (#41668)

Co-authored-by: Computer Network Investigation <121175071+JSCU-CNI@users.noreply.github.com>

kev-the-dev committed:
Change confirmation text (#41650) (#41679)

amoghrajesh committed:
Adding url sanitisation for extra links (#41665) (#41680)

(cherry picked from commit 6c463b3c37fd9aef56ceace7e7b141d892b57054)

uranusjr committed:
Splitting syspath preparation into stages (#41672) (#41694)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

obarisk committed:
fix log for notifier(instance) without __name__ (#41591) (#41699)

Co-authored-by: obarisk <obarisk@yiyixiu.idv.tw>
Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

amoghrajesh committed:
Adding rel property to hyperlinks in logs (#41696) (#41783)

* Adding rel property to hyperlinks in logs

* fixing tests

(cherry picked from commit 79db243d03cc4406290597ad400ab0f514975c79)

jscheffl committed:
Bump micromatch from 4.0.5 to 4.0.8 in /airflow/www (#41755)

Bumps [micromatch](https://github.com/micromatch/micromatch) from 4.0.5 to 4.0.8.
- [Release notes](https://github.com/micromatch/micromatch/releases)
- [Changelog](https://github.com/micromatch/micromatch/blob/4.0.8/CHANGELOG.md)
- [Commits](https://github.com/micromatch/micromatch/compare/4.0.5...4.0.8)

---
updated-dependencies:
- dependency-name: micromatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
(cherry picked from commit e02052b07787e747a78938e60c206d324ef57e7e)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

potiuk committed:
chore(docs): add an example for auth with keycloak (#41687) (#41791)

* chore(docs): add an example for auth with keycloak

Added a new section in the authentication documentation that provides a code of configuring Airflow to work with Keycloak.

* chore(docs): add an example for auth with keycloak

Fix spelling and styling

* chore(docs): add an example for auth with keycloak

Fix static checks

Co-authored-by: Natsu <34879762+hoalongnatsu@users.noreply.github.com>

potiuk committed:
Remove deprecation warning for cgitb in Plugins Manager (#41732) (#41793)

(cherry picked from commit 83ba17f41ee1f88c041289e8b88cf815bc61de7e)

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

jedcunningham committed:
Don't Fail LocalTaskJob on heartbeat (#41704) (#41810)

* Never fail an ltj over a heartbeat

* Log a warning on failed heartbeat

* Avoid using f-string in log

* Remove unnecessary pass statement

(cherry picked from commit 6647610a8e8e3de4d2bfb701e16d1c7b42edd3f8)

Co-authored-by: Collin McNulty <collin.mcnulty@gmail.com>

joaopamaral committed:
Keep FAB compatibility for versions before 1.3.0 in 2.10 (#41549) (#41809)

* Fix: Keep compatibility with old FAB versions (#41549)

* Fix: Tests after #41549 (Keep compatibility with old FAB versions)

* Fix test_dag and test_dagbag

potiuk committed:
Pin universal-pathlib to 0.2.2 as 0.2.3 generates static code check errors (#41715) (#41820)

(cherry picked from commit 1c53961fd4cc1f9085680bf22fbe7f57e29948d4)

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

utkarsharma2 committed:
Fix: DAGs are not marked as stale if the dags folder change (#41433) (#41829)

* Fix: DAGs are not marked as stale if the AIRFLOW__CORE__DAGS_FOLDER changes

* Update airflow/dag_processing/manager.py

* Add testcase

* Add code comment

* Update code comment

* Update the logic for checking the current dag_directory

* Update testcases

* Remove unwanted code

* Uncomment code

* Add processor_subdir when creating processor_subdir

* Fix test_retry_still_in_executor test

* Remove config from test

* Update airflow/dag_processing/manager.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update if condition for readability

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit 9f30a41874454696ae2b215b2d86cb9a62968006)

Lee-W committed:
Set end_date and duration for triggers completed with end_from_trigger as True. (#41834)

Co-authored-by: Karthikeyan Singaravelan <tir.karthi@gmail.com>

shahar1 committed:
logout link in no roles error page fix (#41813) (#41845)

* logout link in no roles error page fix

* review comments

(cherry picked from commit 032ac87b1d93abc53d3281313c957708017e21d4)

Co-authored-by: Gagan Bhullar <g.deepsingh1@gmail.com>

utkarsharma2 committed:
Handle Example dags case when checking for missing files (#41856) (#41874)

Earlier PR create to address the issue was not handling the case for the Example Dags, due to which the example dags were marked as stale since they are not present in the dag_directory. This PR handles that scenarios and update the testcase accordingly.

related: #41432
(cherry picked from commit 435e9687b0c56499bc29c21d3cada8ae9e0a8c53)

potiuk committed:
Bump webpack from 5.76.0 to 5.94.0 in /airflow/www (#41864) (#41879)

Bumps [webpack](https://github.com/webpack/webpack) from 5.76.0 to 5.94.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.76.0...v5.94.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
(cherry picked from commit e8888fe055ac8c341e2fa6631ff6ac5f089d54c5)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

utkarsharma2 committed:
Update version to 2.10.1.

utkarsharma2 committed:
Update RELEASE_NOTES.rst

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,40 @@
+Bug Fixes
+"""""""""
+- Handle Example dags case when checking for missing files (#41856)
+- logout link in no roles error page fix (#41813)
```
```suggestion
- Fix logout link in "no roles" error page(#41813)
```

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,40 @@
+- Don't Fail LocalTaskJob on heartbeat (#41704)
+- Remove deprecation warning for cgitb in Plugins Manager (#41732)
+- Fix log for notifier(instance) without __name__ (#41591)
+- Splitting syspath preparation into stages (#41672)
```
```suggestion
- Splitting syspath preparation into stages (#41694)
```
Should point to the backport

ephraimbuddy reviewed (commented):
Some of the generated changelog PR numbers are pointing to the main branch instead of v2-10-test branch.

potiuk committed:
Adding tojson filter to example_inlet_event_extra example dag (#41873) (#41890)

(cherry picked from commit 87a4a5137519448cfff85fc506b9267ec3e18364)

Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

utkarsharma2 committed:
Skip test_scan_stale_dags_when_dag_folder_change in DB isolation mode (#41893) (#41895)

Since the similar test(test_scan_stale_dags_standalone_mode) are skipped in DB isolation mode

(cherry picked from commit 07af14ae75820a98b60ecffa2949ef7ad70bacab)

jscheffl committed:
Make Scarf usage reporting in major+minor versions and counters in buckets (#41900)

(cherry picked from commit 5a045196704943ef3195f3eeaacf2adae30e5ec1)

ephraimbuddy committed:
Update release command for Airflow 2 (#41907)

* Update release command for Airflow 2

For airflow 2, we create a branch out of v2-10-test and sync that branch
with v2-10-stable to make a release.

* Update CI check

potiuk committed:
Upgrade UV to 0.4.1 (#41925) (#41926)

(cherry picked from commit 9cfc6dfe6eac9bc241783ca580a26875cac742ac)

potiuk committed:
Add backcompat check for executors that don't inherit BaseExecutor (#41906) (#41927)

(cherry picked from commit f168d0aedbc73b86b1bb0e7fb38cad4ce5a66621)

Co-authored-by: Daniel Standish <15932138+dstandish@users.noreply.github.com>

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,43 @@
+- Fix logout link in "no roles" error page (#41845)
+- Set end_date and duration for triggers completed with end_from_trigger as True. (#41834)
+- DAGs are not marked as stale if the dags folder change (#41829)
+- Keep FAB compatibility for versions before 1.3.0 in 2.10 (#41809)
```
```suggestion
- Fix compatibility with FAB provider versions <1.3.0 (#41809)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,43 @@
+- Bump webpack from 5.76.0 to 5.94.0 in /airflow/www (#41879)
+- Adding rel property to hyperlinks in logs (#41783)
+- Field Deletion Warning when editing Connections (#41504)
+- Remove Debian bullseye support (#41569)
```
This is docs only for core. Which makes me wonder, there are no other docs-only changes for 2.10.1?

utkarsharma2 replied:
Moved it to the Docs section. Also, there was a [PR](https://github.com/apache/airflow/commit/83aca021edb9054f43451171c7f415d5a6205536) which was not picked at all by `./dev/airflow-github changelog v2-3-stable v2-3-test` command, that too was docs only change.

utkarsharma2 replied:
I have gone through other changes and I think we have two docs changes which I have listed under the Docs changes sections.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,43 @@
 
 .. towncrier release notes start
 
+Airflow 2.10.1 (2024-08-30)
```
Don't forget to update this date!

utkarsharma2 replied:
Updated it to 2024-09-05, assuming we are doing rc1 today.

potiuk committed:
Protect against None components of universal pathlib xcom backend (#41921) (#41938)

fixes: #41723
(cherry picked from commit 7a75f0a2bc6f964a943db98946ce652c43942180)

potiuk committed:
Lower down universal-pathlib minimum to 0.2.2 (#41939) (#41943)

Beacause we pinned universal-pathlib to 0.2.2 in 2.10.0 adding >=
0.2.3 conflicts with currently released PyPI packages. We shoudl
lower it down in order to allow PyPI constraints to not be
conflicting

(cherry picked from commit 351961caf46e32f10e61691ca6d88d55db1e4d70)
